File: backend/src/main/java/Dominio/Entity/Producto.java
package Dominio.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "Productos")
@Data
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idProducto;

    private String nombreProducto;

    @Column(length = 1000)
    private String descripcion;

    private int stock;
    private double precio;
    private String categoria;

    @Column(columnDefinition = "TEXT")
    private String imagenUrl;

    private boolean activo = true;
}

File: frontend/lib/models/producto.dart
class Producto {
  final int id;
  final String nombre;
  final String descripcion;
  final int stock;
  final double precio;
  final String categoria;
  final String? imagenUrl;
  final bool activo;

  const Producto({
    required this.id,
    required this.nombre,
    required this.descripcion,
    required this.stock,
    required this.precio,
    required this.categoria,
    this.imagenUrl,
    required this.activo,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      id: json['idProducto'] as int,
      nombre: json['nombreProducto'] as String? ?? '',
      descripcion: json['descripcion'] as String? ?? '',
      stock: json['stock'] as int? ?? 0,
      precio: (json['precio'] as num?)?.toDouble() ?? 0.0,
      categoria: json['categoria'] as String? ?? '',
      imagenUrl: json['imagenUrl'] as String?,
      activo: json['activo'] as bool? ?? true,
    );
  }
}